test/datasets/common.py

```diff
@@ -0,0 +1,25 @@
+from parameterized import parameterized
+from torch.utils.data.graph import traverse
+from torch.utils.data.graph_settings import get_all_graph_pipes
+from torchdata.datapipes.iter import Shuffler, ShardingFilter
+from torchtext.datasets import DATASETS
+
+from ..common.torchtext_test_case import TorchtextTestCase
+
+
+class TestShuffleShardDatasetWrapper(TorchtextTestCase):
+    # Note that for order i.e shuffle before sharding, TorchData will provide linter warning
+    # Modify this test when linter warning is available
+    @parameterized.expand(list(DATASETS.items()))
```

Comment:
QQ: it looks like we're not using the `dataset_name` anywhere. Why don't we just pass in the `dataset_fn` to the test by doing something like

```python
@parameterized.expand(list(DATASETS.values()))
```

Reply:
Ahh my bad, thanks for the catch. Will fix it!

Reply:
I actually realized that the `parameterized.expand` decorator complains when passing in the list of  `dataset_fn`. Lmk if you're able to figure out how to resolve the error

Reply:
Yupp, the problem is we need to pass tuples inside list. Just created PR to fix it https://github.com/pytorch/text/pull/1733

Reply:
Awesome. Just incorporated this in my PR https://github.com/pytorch/text/pull/1732